pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -289,10 +297,19 @@ func (c *Controller) Run(stopCh <-chan struct{}) error {
 	// Use NonSlidingUntil so that normal reconnection (disconnected after
 	// running a while) can reconnect immediately while abnormal reconnection
 	// won't be too aggressive.
+
 	go wait.NonSlidingUntil(c.appliedToGroupWatcher.watch, 5*time.Second, stopCh)
 	go wait.NonSlidingUntil(c.addressGroupWatcher.watch, 5*time.Second, stopCh)
 	go wait.NonSlidingUntil(c.networkPolicyWatcher.watch, 5*time.Second, stopCh)
 
+	if !c.restartSynced {
```

Comment:
`Run` will only be called once, I think it's the workers need to check this flag and decide whether to `Done`, and note the case a worker is restarted multiple times while the others never connect to API successfully.

Reply:
I assume meant watchers? Fixed this by adding the restartSynced as a field for watchers as well. To solve the second issue you raised, I separated the restartFullSyncGroup into three different waitgoups for each watcher.